python/pyspark/ml/util.py

```diff
@@ -760,3 +762,127 @@ def _get_active_session(is_remote: bool) -> SparkSession:
     if spark is None:
         raise RuntimeError("An active SparkSession is required for the distributor.")
     return spark
+
+
+class FunctionPickler:
+    """ 
+        This class provides a way to pickle a function and its arguments.
+        It also provides a way to create a pytorch script that can run a
+        function with arguments if they have pickled to a file.
+    """
+    @staticmethod
+    def pickle_func_and_get_path(train_fn: Callable, file_path: str, save_dir: str, *args, **kwargs) -> str:
+        """
+            Given a training function and args, this function will pickle them to a file. 
+
+            Parameters
+            ----------
+            train_fn: Callable
+                The picklable function that will be pickled to a file.
+
+            file_path: str
+                The path where to save the pickled function, args, and kwargs. If its the 
+                empty string, the function will decide on a random name.
+
+            save_dir: str
+                The directory in which to save the file with the pickled function and arguments.
+                Does nothing if the path is specified. If both file_path and save_dir are empty,
+                the function will write the file to the current working directory with a random 
+                name.
+
+            *args: Any
+                Arguments of train_fn  that will be pickled.
+
+            **kwargs: Any
+                Key word arguments to train_fn that will be pickled.
+
+            Returns
+            -------
+            str:
+                The path to the file where the function and arguments are pickled.
+        """
+        if file_path != "":
+            with open(file_path, "wb") as f:
+                cloudpickle.dump((train_fn, args, kwargs), f)
+                return f.name
+
+        if save_dir == "":
+            save_dir = os.getcwd()
+
+        with tempfile.NamedTemporaryFile(dir=save_dir, delete=False) as f:
+            cloudpickle.dump((train_fn, args, kwargs), f)
+            return f.name
+
+    @staticmethod
+    def create_training_script_from_func(fn_output_save_path: str, training_script_save_path: str, pickled_fn_path: str, prefix_code:str ="", suffix_code:str = "") -> str:
```

Comment:
I would name this "create_fn_run_script" or something like that. I would also reorder the args to reflect their order of appearance:

1) pickled_fn_path is loaded
2) fn_output_save_path (rename this to fn_output_path) is written with output
3) training_script_save_path (rename this to script_path) is written with the script

Reply:
Done